examples/samples/session.ts

```diff
@@ -0,0 +1,85 @@
+import {
+  generateUuid,
+  QueueClient,
+  SendableMessageInfo,
+  OnSessionMessage,
+  OnError,
+  MessagingError,
+  delay,
+  ServiceBusMessage,
+  ReceiveMode,
+  Namespace,
+  MessageSession
+} from "../../lib";
+import * as dotenv from "dotenv";
+dotenv.config();
+
+const str = process.env.SERVICEBUS_CONNECTION_STRING || "";
+const path = process.env.QUEUE_NAME || "";
+console.log("str: ", str);
+console.log("path: ", path);
+
+let ns: Namespace;
+async function main(): Promise<void> {
+  ns = Namespace.createFromConnectionString(str);
+  const client = ns.createQueueClient(path, { receiveMode: ReceiveMode.peekLock });
+
+  await sendMessages(client, "session-1");
+  await sendMessages(client, "session-2");
+  await sendMessages(client, "session-3");
+  await sendMessages(client, "session-4");
+
+  await receiveMessages(client);
+
+  await client.close();
+}
+
+async function sendMessages(queueClient: QueueClient, sessionId: string): Promise<void> {
+  const data = [
+    { step: 1, title: "Shop" },
+    { step: 2, title: "Unpack" },
+    { step: 3, title: "Prepare" },
+    { step: 4, title: "Cook" },
+    { step: 5, title: "Eat" }
+  ];
+
+  for (let index = 0; index < data.length; index++) {
+    const element = data[index];
+    const message: SendableMessageInfo = {
+      sessionId: sessionId,
+      body: `${element.step} ${element.title}`,
+      label: "RecipeStep",
+      timeToLive: 2 * 60 * 1000, // 2 minutes
+      messageId: generateUuid()
+    };
+
+    console.log(`Message sent: ${message.body} SessionId : ${sessionId}`);
+    await queueClient.send(message);
+  }
+}
+
+async function receiveMessages(queueClient: QueueClient): Promise<void> {
+  const onMessage: OnSessionMessage = async (
+    messageSession: MessageSession,
+    brokeredMessage: ServiceBusMessage
+  ) => {
+    console.log(
+      `Message received: ${brokeredMessage.body} SessionId : ${brokeredMessage.sessionId}`
+    );
+    await brokeredMessage.complete();
+  };
+  const onError: OnError = (err: MessagingError | Error) => {
+    console.log(">>>>> Error occurred: ", err);
+  };
+  queueClient.receiveMessgesFromSessions(onMessage, onError);
+  await delay(10000);
+}
+
+main()
+  .then(() => {
+    console.log(">>>> Calling close....");
+    return ns.close();
```

Comment:
@ramya-rao-a suggested adding a try-catch in main and invoke ns.close() once in the finally block - https://github.com/Azure/azure-service-bus-node/pull/127/files

Reply:
@ramya0820 suggested this comment, @ramya-rao-a is on vacations.

@HarshaNalluru  @ramya0820 I was following the same pattern as old samples. I think before changing/merging anything we should wait for @ramya-rao-a suggestion.
I don't see any additional benefit of adding extra try-catch-finally block in a separate method. I agree that we should close namespace/queueClient etc before exiting but instead of adding new method , we can write 
```
main()
  .catch((err) => {
    console.log("error: ", err);
  })
  .then(() => {
    console.log(">>>> Calling close....");
    return ns.close();
  });
```
It would give us the same result as try, catch and finally.
but still I would wait for @ramya-rao-a @AlexGhiondea suggestions on this.

Reply:
It looks cleaner this way. Sure, let's wait for @ramya-rao-a.

Reply:
Chaining .then() is definitely not the same as using a finally block (such as on force quitting programs).

Actually, it looks like we can use .finally() on a promise
https://blog.mariusschulz.com/2018/11/27/using-promise-prototype-finally-in-typescript

I think this would be our best bet.

Reply:
I am missing something here... Can someone point me to the original comment?

Reply:
In this way, we have to write `ns.close()` twice ?
```
main()
  .then(() => {
    console.log(">>>> Calling close....");
    return ns.close();
  })
  .catch((err) => {
    console.log("error: ", err);
    ns.close();
  });`

Reply:
Yes, but the first one would be inside the `main()` call right after we close the client which would seem more natural.

As a user, I am done working with the client, so I call close on it. I am done with the namespace, so I call close on it. One after the other.

The `main().catch()` would be for unhandled error scenarios